File: c/linear_search.c
#include <stdbool.h>
#include <stdio.h>

bool linear_search(int *haystack, size_t size, int needle) {
    for (size_t i = 0; i < size; ++i) {
        if (haystack[i] == needle) {
            return true;
        }
    }

    return false;
}

int main(void) {
    int arr[] = {1, 3, 5, 10, 3, 2, 7};

    size_t size = sizeof(arr) / sizeof(*arr);

    printf("%d\n", linear_search(arr, size, 5));
    printf("%d\n", linear_search(arr, size, 10));
    printf("%d\n", linear_search(arr, size, 7));
    printf("%d\n", linear_search(arr, size, -3));

    return 0;
}

File: typescript/linear_search.ts
function linear_search(haystack: number[], needle: number): boolean {
    for (let i = 0; i < haystack.length; ++i) {
        if (haystack[i] === needle) {
            return true;
        }
    }
    return false;
}

const list = [1, 3, 5, 6, 9, 10, 44, 2];

console.log(linear_search(list, 5))
console.log(linear_search(list, 10))
console.log(linear_search(list, 0))
console.log(linear_search(list, -11))
console.log(linear_search([], -11))
